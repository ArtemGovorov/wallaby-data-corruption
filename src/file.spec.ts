import * as fs from "fs";
import * as crypto from "crypto";
import { join } from "path";

it("should have the right checksum", (done) => {
    const path = join(__dirname, "../files/sqllite.db");
    var md5sum = crypto.createHash('md5');
    var stream = fs.createReadStream(path);

    stream.on("data", (data: any) => md5sum.update(data));
    stream.on("end", () => {
        const digest = md5sum.digest("hex");

        expect(digest).toMatchSnapshot();
        done();
    });
});