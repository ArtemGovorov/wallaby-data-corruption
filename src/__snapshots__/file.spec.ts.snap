// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have the right checksum 1`] = `"f1af7219401c068adb570f7bd5095dab"`;
